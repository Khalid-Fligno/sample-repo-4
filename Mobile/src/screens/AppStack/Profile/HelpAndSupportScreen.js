import React from 'react';
import {
  StyleSheet,
  SafeAreaView,
  View,
  Text,
  ScrollView,
  Dimensions,
  Platform,
} from 'react-native';
import { Linking } from 'expo';
import Loader from '../../../components/Shared/Loader';
import CustomButton from '../../../components/Shared/CustomButton';
import colors from '../../../styles/colors';
import fonts from '../../../styles/fonts';
import globalStyle from '../../../styles/globalStyles';
import ProfileStyles from './ProfileStyles';
const { width } = Dimensions.get('window');

export default class HelpAndSupportScreen extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      loading: false,
    };
  }
  goToHelp = () => {
    Linking.openURL('https://fitazfk.zendesk.com/hc/en-us');
  }

  goToReport = () => {
    Linking.openURL('https://fitazfk.canny.io/fitazfk-app-feedback');
  }
  render() {
    const { loading } = this.state;
    return (
      <SafeAreaView style={globalStyle.safeContainer}>
        <View style={[globalStyle.container, { paddingHorizontal: 0, alignItems: 'center' }]}>
          <ScrollView contentContainerStyle={[ProfileStyles.scrollView, { padding: 0 }]}>
            <View style={ProfileStyles.scrollView}>
              <Text style={ProfileStyles.header}>
                Help & Support
              </Text>
              <Text style={ProfileStyles.paragraph}>
                Follow this link to see our frequently asked questions, or submit your own!
              </Text>
            </View>
            <View style={ProfileStyles.buttonContainer}>
              <View>
                <CustomButton
                  title="Go to FAQ's Page"
                  onPress={() => this.goToHelp()}
                />
              </View>
              <View style={{ marginTop: 10 }}>
                {
                  Platform.OS === 'android' ?
                    <CustomButton
                      title="Report an issue"
                      onPress={() => this.goToReport()}
                    />
                    :
                    <CustomButton
                      title="Report an issue"
                      onPress={() => this.goToReport()}
                    />
                }
              </View>
            </View>
          </ScrollView>
          <Loader
            loading={loading}
            color={colors.charcoal.standard}
          />
        </View>
      </SafeAreaView>
    );
  }
}
