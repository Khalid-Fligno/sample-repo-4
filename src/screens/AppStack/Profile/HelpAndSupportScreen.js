import React from 'react';
import { StyleSheet, SafeAreaView, View, Text, ScrollView, Dimensions } from 'react-native';
import Loader from '../../../components/Shared/Loader';
import colors from '../../../styles/colors';
import fonts from '../../../styles/fonts';

const { width } = Dimensions.get('window');

export default class HelpAndSupportScreen extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      loading: false,
    };
  }
  render() {
    const { loading } = this.state;
    return (
      <SafeAreaView style={styles.safeContainer}>
        <View style={styles.container}>
          <ScrollView contentContainerStyle={styles.scrollView}>
            <Text style={styles.header}>
              Help & Support
            </Text>
            <Text style={styles.paragraph}>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
            </Text>
          </ScrollView>
          <Loader
            loading={loading}
            color={colors.charcoal.standard}
          />
        </View>
      </SafeAreaView>
    );
  }
}

const styles = StyleSheet.create({
  safeContainer: {
    flex: 1,
    backgroundColor: colors.black,
  },
  container: {
    flex: 1,
    backgroundColor: colors.white,
    alignItems: 'center',
  },
  scrollView: {
    width,
    padding: 15,
  },
  header: {
    fontFamily: fonts.bold,
    fontSize: 24,
    marginBottom: 10,
  },
  paragraph: {
    fontFamily: fonts.standard,
    fontSize: 14,
  },
});
