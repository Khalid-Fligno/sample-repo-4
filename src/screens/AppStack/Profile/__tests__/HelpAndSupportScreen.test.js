import React from 'react';
import renderer from 'react-test-renderer';
import HelpAndSupportScreen from '../HelpAndSupportScreen';

describe('HelpAndSupportScreen', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<HelpAndSupportScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
