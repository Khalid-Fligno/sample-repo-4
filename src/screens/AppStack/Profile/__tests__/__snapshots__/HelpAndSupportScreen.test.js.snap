// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpAndSupportScreen renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "rgb(0, 0, 0)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(255, 255, 255)",
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "width": 750,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "padding": 15,
              "width": 750,
            }
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "GothamBold",
                "fontSize": 24,
                "marginBottom": 10,
              }
            }
          >
            Help & Support
          </Text>
          <Text
            style={
              Object {
                "fontFamily": "GothamBook",
                "fontSize": 14,
              }
            }
          >
            Follow this link to see our frequently asked questions, or submit your own!
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginBottom": 10,
              "marginTop": 10,
              "width": 750,
            }
          }
        >
          <View
            style={
              Object {
                "borderRadius": 3,
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "rgb(20, 20, 20)",
                    "borderColor": "#2089dc",
                    "borderRadius": 4,
                    "borderWidth": 0,
                    "flexDirection": "row",
                    "height": 45,
                    "justifyContent": "center",
                    "padding": 8,
                    "shadowColor": "rgb(38, 38, 40)",
                    "shadowOffset": Object {
                      "height": 2,
                      "width": 0,
                    },
                    "shadowOpacity": 0.8,
                    "shadowRadius": 2,
                    "width": 730,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "rgb(255, 255, 255)",
                      "fontFamily": "GothamBold",
                      "fontSize": 14,
                      "marginTop": 3,
                      "paddingBottom": 1,
                      "paddingTop": 2,
                      "textAlign": "center",
                    }
                  }
                >
                  GO TO FITAZFK HELP
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
